### DISCUSSION TITLE
Inconsistencies between data validations (date/time)

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#10527] [Help]
I have a MySQL table already in use by my application.
It contains several date fields but the validation is different between the fields!

My first column (for example: buy date) was created more than a year ago.
My second column (for example: return date) was created yesterday.
The configuration is strictly identical:
![image](https://github.com/Budibase/budibase/assets/91942877/739cfc23-ba01-4837-aafd-532dd81a5e99)

When I add a row via the data tab, I get a validation error ('return_date must be a valid date') for the return date column if the field remains empty. No error for the same lack of value in the purchase date column (NULL is a valid datetime value if the field isn't required). 
This should return no errors in either case or errors in all cases.

Has there been a change to the validation of value types recently?
I feel like I have other similar problems that may be related to a change like this (with number columns).

Thank you!

**Comment by diogenesbrussels** [comment]
Further analysis:
`"date_buy":{"name":"date_buy","autocolumn":false,"type":"datetime","constraints":{"presence":false}}`

`"date_return":{"type":"datetime","constraints":{"type":"string","length":{},"presence":false,"datetime":{"latest":"","earliest":""}},"name":"date_return","ignoreTimezones":null}`

The definition isn't the same, why is there a contraint on type string?

**Reply by diogenesbrussels** [reply]
I continue to feed with another similar problem:
I have a simple MySQL table with user data: name (text), email (text), active (boolean), assigned (boolean).
When adding a row, no problem.
When I want to modify the row I just created:
![image](https://github.com/Budibase/budibase/assets/91942877/60a24ee9-7ec6-4d5b-9827-11b2688b65ec)

Why? 
The field type is validated by budibase, the field is a toggle created by budibase, how can it contain an invalid value?

I hope to understand and solve this problem because it creates unexpected problems with things that should be so simple :/

**Reply by diogenesbrussels** [reply]
Seems similar to #10575 for the boolean validation
Is the problem the same for datetime?

**Reply by FlaminWrap** [reply]
Hey, thought I would chime in and say I have just experienced the same issue.

The app and MySQL database was created on Budibase 2.9.38. If you change the boolean value, it works fine.

![image](https://github.com/Budibase/budibase/assets/97764630/84648f35-0b3f-419e-ae03-bc2e8b1548d4)
